### ISSUE TITLE
Sugar for simple `destruct`

### ISSUE THREAD

**Issue by meyercm** [mgwidmann/elixir-pattern_tap#4]
I've found myself using `destruct` just save a variable mid-pipeline, without needing to destructure anything; e.g.

```elixir
items
|> Enum.map(&fix_common_mistakes/1)
|> destruct(fixed_items ~> fixed_items)
|> Items.save!

#now use fixed_items rather than re-correct the original list
```
I'd like syntax to allow less typing:

```elixir
items
|> Enum.map(&fix_common_mistakes/1)
|> extract(fixed_items) # only type it once!
|> Items.save!
```

Making the 2-arity version of `destruct` seems like a simple change, but now the name is misleading, hence using `extract`.  How would you feel about adding this functionality?  I don't mind writing up a pull request if you are open to the idea.

**Comment by mgwidmann** [comment]
Sure. Its basically leaking the variable to the outside scope (which is your intention). Though, `extract` isn't the **best** term... any others you can think of? If not we can just go with that...

Thanks!

**Comment by meyercm** [comment]
Others that I thought of and rejected previously included: `save_var` and `create_variable`, which are too long; and `var`, which lacks explicitness.

Reading your last comment, I like the words you used: `leak`, or perhaps `leak_as`.  Not only does it accurately describe what's going on, it's in keeping with the pipe metaphor. `assign`, `retain`, `preserve` and `keep` I think provide good description as well.

The real test for me is how it looks in code.  I like all of these, possibly `assign_to` a little more than the others now that its in a pipeline. Do any of these appeal to you?

```elixir
stuff
|> operation_1
|> leak(partial_result)
|> operation_2
|> preserve(partial_result_2)
|> operation_3
|> keep_as(partial_result_3)
|> operation_4
|> retain(partial_result_4)
|> operation_5
|> assign_to(result)
```

**Comment by mgwidmann** [comment]
I like `leak`, its short and descriptive.

**Comment by mgwidmann** [comment]
Closing since #5 was merged 🎉

**Comment by meyercm** [comment]
Is publishing to Hex now an option, or are you waiting to merge in more changes?

**Comment by mgwidmann** [comment]
Published `v0.4.0`, just had to find time to do it.
